the-collab-lab/tcl-78-smart-shopping-list
# Rt tc sort shopping list items

|   | Type |
| --- | ---|
| ✓ | New Feature |
| ✓ | Items are displayed by urgency |


## Description



The items are now sorted in each individual list by purchase urgency. This means that the items are now displayed in descending order of  how soon the user needs to buy each of them again. For this, we made the following changes:

`firebase.js`

We created the `comparePurchaseUrgency` function where we created an urgency index, which measures the time between today and the `dayNextPurchased` property. We also created the inactive index, which measures the time between today and the `dateLastPurchased` property of the item.

Then we assign a category (soon, kind of soon, not soon, overdue and inactive) based on those two indexes and then sort the items by urgency and alphabetically.

`List` view

Items are displayed in sub-lists, depending on their category.

## Related Issue

closes issue #13 

## Acceptance Criteria


If you complete all of the previous acceptance criteria, consider what happens when an item’s `dateNextPurchased` has passed, but it isn’t yet inactive. Let’s call that item “overdue”.


## Type of Changes

`New feature`

## Updates

### Before

![Captura de pantalla 2024-09-20 a la(s) 18 14 15](https://github.com/user-attachments/assets/d12b9182-1e3f-4339-ab87-1ff2f32d8232)


### After

![Captura de pantalla 2024-09-20 a la(s) 18 15 09](https://github.com/user-attachments/assets/d22e83f4-16c6-4cf6-9936-7910fc487439)


## Testing Steps / QA Criteria

- select a shopping list that you did not create (e.g. `test-list`) 
- add an item to the list in the ManageList view
- return to the List view and check the item is in the right category
- change the lastDatePurchased to longer than 30 days
- return to the List view and check the item is in the overdue category (top)
